### PR TITLE
add trycatch to renderer.tok() call

### PR DESCRIPTION
**Pull request by sw-yx**
**Marked version:**

	0.7.0

**Markdown flavor: n/a

## Description

- Fixes https://github.com/markedjs/marked/issues/1547#issuecomment-531360288

## Contributor

- [ ] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [x] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by sw-yx** [comment]
diff looks big bc of indentation but really i only added the try catch

File: lib/marked.js [hunk 1]
```diff
@@ -1186,125 +1186,130 @@ Parser.prototype.parseText = function() {
...
+        var errMsg = 'Token with "' + this.token.type + '" type was not found.';
+        if (this.options.silent) {
+          console.log(errMsg);
+        } else {
+          throw new Error(errMsg);
+        }
       }
     }
+  } catch (err) {
+    console.error(err) // let developer see stacktrace
```
Comment (UziTech) [inline lib/marked.js]: I feel like we still need to throw the same error so users can log it how they want. We could just change the `err.message` and rethrow it like on [line 1651](https://github.com/markedjs/marked/blob/master/lib/marked.js#L1651)

**Comment by UziTech** [comment]
This will need to be rebased and the changes will need to be moved to the `/src/` folder.

**Comment by abrahamcuenca** [comment]
What is the status of this PR?

**Comment by UziTech** [comment]
@sw-yx do you still want to work on this PR?

**Comment by sw-yx** [comment]
hi! not really. i dont even remember what i used this for, haha. can revisit if i ever have this issue again. sorry for the long silence

**Pull request closed without merging**